buffalo/cmd/new.go

```diff
@@ -121,24 +136,38 @@ func notInGoPath(ag newapp.Generator) error {
-	newCmd.Flags().BoolVar(&app.WithDep, "with-dep", false, "adds github.com/golang/dep to your app")
-	newCmd.Flags().BoolVar(&app.SkipWebpack, "skip-webpack", false, "skips adding Webpack to your app")
-	newCmd.Flags().BoolVar(&app.SkipYarn, "skip-yarn", false, "use npm instead of yarn for frontend dependencies management")
-	newCmd.Flags().StringVar(&app.DBType, "db-type", "postgres", fmt.Sprintf("specify the type of database you want to use [%s]", strings.Join(pop.AvailableDialects, ", ")))
-	newCmd.Flags().StringVar(&app.Docker, "docker", "multi", "specify the type of Docker file to generate [none, multi, standard]")
-	newCmd.Flags().StringVar(&app.CIProvider, "ci-provider", "none", "specify the type of ci file you would like buffalo to generate [none, travis, gitlab-ci]")
-	newCmd.Flags().StringVar(&app.VCS, "vcs", "git", "specify the Version control system you would like to use [none, git, bzr]")
-	newCmd.Flags().IntVar(&app.Bootstrap, "bootstrap", app.Bootstrap, "specify version for Bootstrap [3, 4]")
+	newCmd.Flags().Bool("api", false, "skip all front-end code and configure for an API server")
+	newCmd.Flags().BoolP("force", "f", false, "delete and remake if the app already exists")
+	newCmd.Flags().BoolP("verbose", "v", false, "verbosely print out the go get commands")
+	newCmd.Flags().Bool("skip-pop", false, "skips adding pop/soda to your app")
+	newCmd.Flags().Bool("with-dep", false, "adds github.com/golang/dep to your app")
+	newCmd.Flags().Bool("skip-webpack", false, "skips adding Webpack to your app")
+	newCmd.Flags().Bool("skip-yarn", false, "use npm instead of yarn for frontend dependencies management")
+	newCmd.Flags().String("db-type", "postgres", fmt.Sprintf("specify the type of database you want to use [%s]", strings.Join(pop.AvailableDialects, ", ")))
+	newCmd.Flags().String("docker", "multi", "specify the type of Docker file to generate [none, multi, standard]")
+	newCmd.Flags().String("ci-provider", "none", "specify the type of ci file you would like buffalo to generate [none, travis, gitlab-ci]")
+	newCmd.Flags().String("vcs", "git", "specify the Version control system you would like to use [none, git, bzr]")
+	newCmd.Flags().Int("bootstrap", 3, "specify version for Bootstrap [3, 4]")
+	viper.BindPFlags(newCmd.Flags())
+	cobra.OnInitialize(initConfig)
+	newCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.buffalo.yaml)")
+}
+
+func initConfig() {
+	if cfgFile != "" { // enable ability to specify config file via flag
+		viper.SetConfigFile(cfgFile)
+	} else {
+		viper.SetConfigName(".buffalo") // name of config file (without extension)
+		viper.AddConfigPath("$HOME")    // adding home directory as first search path
+		viper.AutomaticEnv()            // read in environment variables that match
+	}
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err == nil {
+		fmt.Println("Using config file:", viper.ConfigFileUsed())
+	}
 }
 
```

Comment:
There is no need to init the config file if we don't provide one -- use sane defaults in-memory but toss it in the buffalo project dir, not the home dir if this is a non-negotiable.